### ISSUE TITLE
Document "/media" REST endpoint file upload mechanics

### ISSUE THREAD

**Issue by CoherenceLabs** [WP-API/docs#129]
The REST v2 `/media` [endpoint reference](https://developer.wordpress.org/rest-api/reference/media/#create-a-media-item) would benefit from a mention of file content upload mechanics. The first technique documented in [this gist I found](https://gist.github.com/ahmadawais/0ccb8a32ea795ffac4adfae84797c19a) (set the `Content-Disposition` HTTP header field to `form-data; filename="thefile.jpg"` and send the file's content as the HTTP body data) ended up working for me, but it would be helpful to have official documentation on this embedded in or linked from the `/media` endpoint reference.

**Comment by TimothyBJacobs** [comment]
Yeah this could definitely be improved. Unfortuantely, however, the Reference documentation is entirely generated from the Schema of the nedpoint. So we don't actually have a way right now to add custom content to those pages.

The need for that functionality has come up a couple of times, however. So it's probably worth adding support for that somehow. Just not sure what it would look like at the moment.
